src/components/elements/carousel/carousel.body.tsx

```diff
@@ -0,0 +1,26 @@
+import React, { forwardRef, ReactNode } from 'react'
+import Slides from './carousel.slides'
+import { Container } from './carousel.styles'
+import ViewPort from './carousel.viewport'
+
+export interface ICarouselBody extends React.HtmlHTMLAttributes<HTMLDivElement> {
+    render_nav?: () => ReactNode
+    render_controls?: () => ReactNode
+}
+
+const CarouselBody = (
+    { render_nav, render_controls, children, ...rest }: React.PropsWithChildren<ICarouselBody>,
+    ref: React.ForwardedRef<HTMLDivElement>,
+) => {
+    return (
+        <Container>
+            <ViewPort ref={ref} {...rest}>
+                <Slides>{children}</Slides>
+            </ViewPort>
+            {render_nav?.()}
+            {render_controls?.()}
+        </Container>
+    )
+}
+
+export default forwardRef<HTMLDivElement, React.PropsWithChildren<ICarouselBody>>(CarouselBody)
```

Comment:
```suggestion
const CarouselBody = forwardRef<HTMLDivElement, React.PropsWithChildren<ICarouselBody>>((
    { render_nav, render_controls, children, ...rest },
    ref,
) => {
    return (
        <Container>
            <ViewPort ref={ref} {...rest}>
                <Slides>{children}</Slides>
            </ViewPort>
            {render_nav?.()}
            {render_controls?.()}
        </Container>
    )
})

CarouselBody.displayName = 'CarouselBody'

export default CarouselBody
```
Types for `ref` and `props` are duplicated, We can just provide the types to `forwardRef` generics and let TS infer.

Reply:
thanks, done :pray: